test/directives_test.exs

```diff
@@ -755,6 +755,90 @@ defmodule Surface.DirectivesTest do
              """
     end
   end
+
+  describe ":values" do
+    test "passing a keyword list" do
+      html =
+        render_surface do
+          ~H"""
+          <div :values={{ hello: :world, foo: "bar", one: 2, yes: true }}>
+            Some Text
+          </div>
+          """
+        end
+
+      assert html =~ """
+             <div phx-value-foo="bar" phx-value-hello="world" phx-value-one="2" phx-value-yes="true">
+               Some Text
+             </div>
+             """
+    end
+
+    test "passing a map" do
+      html =
+        render_surface do
+          ~H"""
+          <div :values={{ %{hello: :world, foo: "bar", one: 2, yes: true} }}>
+            Some Text
+          </div>
+          """
+        end
+
+      assert html =~ """
+             <div phx-value-foo="bar" phx-value-hello="world" phx-value-one="2" phx-value-yes="true">
+               Some Text
+             </div>
+             """
+    end
+
+    test "passing unsupported types" do
+      assert_raise(RuntimeError, ~r(invalid value for key ":map" in attribute ":values".), fn ->
+        render_surface do
+          ~H"""
+          <div :values={{ map: %{}, tuple: {} }}>
+            Some Text
+          </div>
+          """
```

Comment:
Can you try using something like:

```elixir
  render_surface do
    ~H"""
    <div :values={{ 
      map: %{}, 
      tuple: {} }}
    >
      Some Text
    </div>
    """
  end
```

This way we can make sure the error is pointing to the right line by using something like:

```elixir
assert_raise(RuntimeError, ~r(code:2: invalid value for key ":map" in attribute ":values".) 
```

Reply:
I believe we only get the code line `code:2` on a `CompileError`, but not on a `RuntimeError`. Am I missing something?

Reply:
Yeah, you're right. Sorry.